### DISCUSSION TITLE
How to access a large object with focusAtom.

### DISCUSSION THREAD

**Discussion by wotjd1799** [pmndrs/jotai#1386] [Q&A] [answered]
Because I am not good at English, I have not found an example of modifying into a large object using a `focusAtom`.

I am currently using it as below.
```Javascript
const Data = {
  state: {
    number: 0,
    name: "",
    ...
  },
  keyword: "",
  user_list: [],
  ...
}

export const dataAtom = atomWithImmer(Data);

//const state = focusAtom(dataAtom, optic.props("state")); <- I only had access to the first element.
```

How can I access and modify the `number`?

**Comment by dai-shi** [comment, marked as answer]
immer doesn't work in this case, but other than that, it should be something like:
```js
const initialData = {
  state: {
    number: 0,
    name: "",
    ...
  },
  keyword: "",
  user_list: [],
  ...
}

const dataAtom = atom(initialData);

const stateAtom = focusAtom(dataAtom, (optic) => optic.props("state"));
// stateAtom works like `atom(initialData.state)`, but linked.
```

**Reply by wotjd1799** [reply]
How can I modify only the `number` that exists inside the state?

**Reply by wotjd1799** [reply]
And I am currently modifying other values in Data from external components through immer.
If i use `immer`, i can access the value, such as draft.state.number, but if i use `focusAtom,` how do i access it?

**Reply by dai-shi** [reply]
Why do you need `focusAtom` in the first place? You can just start with `atomWithImmer(initialData)` without `focusAtom`. Roughly speaking, `focusAtom` and `atomWithImmer` is incompatible. `focusAtom` is functional pure style, where as `atomWithImmer` is mutable impure style.

**Reply by wotjd1799** [reply]
I didn't mean to use `focusAtom` unconditionally, and I thought it was a function worth considering, but I think I made a mistake! Thank you so much

**Reply by wotjd1799** [reply]
If so, when changing the atom using Immer, is there a way to access it at the top level (outside the class)?
https://codesandbox.io/s/quizzical-margulis-y9p0lz?file=/src/App.js
If you don't know my intentions, I'll try to organize them again.

**Reply by dai-shi** [reply]
Something like this? https://codesandbox.io/s/silly-sunset-ynz324

**Reply by wotjd1799** [reply]
God, that's exactly the way I've been looking. I didn't know it would be solved so simply.
